server.js

```diff
@@ -0,0 +1,247 @@
+const http = require('http');
+const fs = require('fs');
+const qs = require('querystring');
+
+const authorizedUsers = {
+  banana: 'peel',
+};
+
+const server = http.createServer(function(req, res) {
+  // GET
+
+  if (req.method === 'GET') {
+    if (req.url === '/') {
+      fs.readFile('./public/index.html', 'utf8', (err, data) => {
+        if (err) {
+          return res.end(`Error: ${err}`);
+        } else {
+          return res.end(data);
+        }
+      });
+    }
+
+    fs.readFile(`./public${req.url}`, 'utf8', (err, data) => {
+      if (err) {
+        return res.end(`Error: ${err}`);
+      } else {
+        return res.end(data);
+      }
+    });
+  }
+
+  // POST
+
+  if (req.method === 'POST' && req.url === '/elements') {
+    if (req.headers.authorization) {
+      const encodedString = req.headers.authorization.slice(6);
+      const base64Buffer = new Buffer(encodedString, 'base64');
+      const decodedString = base64Buffer.toString();
+      let username = decodedString.slice(0, decodedString.indexOf(':'));
+      let password = decodedString.slice(decodedString.indexOf(':') + 1);
+
+      if (authorizedUsers.hasOwnProperty(username) && authorizedUsers[username] === password) {
+        let body = '';
+
+        req.on('data', (chunk) => {
+          body += chunk;
+        });
+
+        req.on('end', (chunk) => {
+          if (chunk) {
+            body += chunk;
+          } else {
```

Comment:
Get rid of the else since if there is a chunk in the end, then none of the code in the else would be executed.